File: sdk-ios/Sources/UXClone/SessionManager.swift
import Foundation

/// Manages session and anonymous ID persistence via UserDefaults.
final class SessionManager {
    private static let anonKey    = "uxclone_anon_id"
    private static let sessionKey = "uxclone_session_id"

    private let startDate: Date
    private let sessionId: String
    private let anonymousId: String

    init() {
        self.startDate = Date()

        // Create or restore anonymous ID
        if let existing = UserDefaults.standard.string(forKey: Self.anonKey) {
            self.anonymousId = existing
        } else {
            let anon = "anon_\(UUID().uuidString)"
            UserDefaults.standard.set(anon, forKey: Self.anonKey)
            self.anonymousId = anon
        }

        // Always create a fresh session ID
        let sid = UUID().uuidString
        UserDefaults.standard.set(sid, forKey: Self.sessionKey)
        self.sessionId = sid
    }

    func getSessionId()  -> String { sessionId }
    func getAnonymousId() -> String { anonymousId }

    func getElapsedMs() -> Int64 {
        Int64(Date().timeIntervalSince(startDate) * 1000)
    }
}

File: sdk-android/library/src/main/kotlin/app/uxclone/sdk/SessionManager.kt
package app.uxclone.sdk

import android.content.Context
import java.util.UUID

/** SharedPreferences-backed session persistence. */
internal class SessionManager(context: Context) {
    companion object {
        private const val PREFS    = "uxclone_prefs"
        private const val ANON_KEY = "anon_id"
    }

    private val prefs     = context.getSharedPreferences(PREFS, Context.MODE_PRIVATE)
    private val startTime = System.currentTimeMillis()
    val sessionId:   String
    val anonymousId: String

    init {
        // Restore or create anonymous ID
        anonymousId = prefs.getString(ANON_KEY, null) ?: run {
            val id = "anon_${UUID.randomUUID()}"
            prefs.edit().putString(ANON_KEY, id).apply()
            id
        }
        // Always fresh session ID
        sessionId = UUID.randomUUID().toString()
    }

    fun getElapsedMs(): Long = System.currentTimeMillis() - startTime
}
